lib/export-task.js

```diff
@@ -18,6 +18,30 @@ class ExportTask {
       this[key] = content[key];
   }
 
+  async retry() {
+    const queryResult = await query(`
+      PREFIX export: <http://mu.semte.ch/vocabularies/ext/export/>
+      SELECT ?retries WHERE {
+        ${sparqlEscapeUri(this.uri)} export:numberOfRetries ?retries
+      }
+    `)
+    const retries = +queryResult.results.bindings[0].retries.value
+    await update(`
+      PREFIX export: <http://mu.semte.ch/vocabularies/ext/export/>
+  
+      WITH <${process.env.MU_APPLICATION_GRAPH}>
+      DELETE {
+        ${sparqlEscapeUri(this.uri)} export:numberOfRetries ?retries.
+      }
+      INSERT { 
+        ${sparqlEscapeUri(this.uri)} export:numberOfRetries ${sparqlEscapeInt(retries+1)}.
+      } WHERE {
+        ${sparqlEscapeUri(this.uri)} export:numberOfRetries ?retries.
+      }
+    `);
+    await this.perform()
```

Comment:
You need a way to indicate that a retry for a task is executing. If the retry period is shorter than the time the execution of a tasks take, a new retry will be started before the previous retry has finished.

In general the microservice also has the constraint to run only 1 export task at the same time (see https://github.com/lblod/mandaten-download-generator-service/blob/cecf98f4bf94a0cb875f2f8b6f75cf1a2e3b23ca/app.js#L32) to limit the load on the database. With the current retry mechanism this constraint may be broken.